### PR TITLE
[16.04] Backport workflow fix

### PR DESCRIPTION
**Pull request by dannon**
This is an extra hammer swing at #1531 / #1542, which @remimarenco encountered.  He said this resolves it.

### CODE REVIEW

**Comment by dannon** [comment]
The kinder, gentler fix in the previous pr (#2505) did not work, so here we are.

File: lib/galaxy/jobs/actions/post.py [hunk 1]
```diff
@@ -427,7 +427,7 @@ def execute(cls, app, sa_session, action, job, replacement_dict):
                 else:
                     log.debug("No job found yet for wfi_step %s, (step %s)" % (wfi_step, wfi_step.workflow_step))
             for j2c in jobs_to_check:
-                creating_jobs = [(x, x.dataset.creating_job) for x in j2c.input_datasets if x.dataset.creating_job]
+                creating_jobs = [(x, x.dataset.creating_job) for x in j2c.input_datasets if x.dataset and x.dataset.creating_job]
```
Comment (nsoranzo) [inline lib/galaxy/jobs/actions/post.py]: Should we add some logging in order to find out what are the "offending" `input_datasets`? E.g.:

``` python
creating_jobs = []
for x in j2c.input_datasets:
    if not x.dataset:
        log.debug("...")
    elif not x.dataset.creating_job:
        log.debug("...")
    else:
        creating_jobs.append((x, x.dataset.creating_job))
```

Comment (dannon) [inline lib/galaxy/jobs/actions/post.py]: @nsoranzo Sure, I can do that.  Someone will need to actually follow up on the debug output though ;)

Comment (nsoranzo) [inline lib/galaxy/jobs/actions/post.py]: Well, @remimarenco has the test case, soooo :stuck_out_tongue_winking_eye:

Comment (remimarenco) [inline lib/galaxy/jobs/actions/post.py]: Haha! Will let you know then :)

**Comment by nsoranzo** [comment]
@galaxybot test this

**Comment by nsoranzo** [comment]
:+1: Thanks @dannon!

**Comment by remimarenco** [comment]
Added in my cloud. We should have a few more informations. @dannon let me know where I should dump the logs ;) (On this thread or?)

**Comment by dannon** [comment]
@remimarenco You can gist and send them to me directly if you see anything interesting, though, to make them interesting you'd have to do some extra work to cross-check which steps and datasets in the workflow/invocation are problematic.

**Pull request merged**